### PR TITLE
Add github actions ci pipeline

### PR DESCRIPTION
**Pull request by mwang2016**

### CODE REVIEW

File: test/config.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ kafka_topic_2: kafka_topic_2
 kafka_header_topic: kafka_header_topic
 kafka_header_index: kafka
 connector_path: /usr/local/share/kafka/plugins
-connector_build_target: /home/circleci/repo/target
+connector_build_target: /usr/local/share/kafka-connector
 kafka_home: /usr/local/kafka
 kafka_connect_home: /home/circleci/repo
-old_connector_name: splunk-kafka-connect-v1.2.0.jar
+old_connector_name: splunk-kafka-connect-v1.2.0.jar
```
Comment (chaitanyaphalak) [inline test/config.yaml:18]: can be changed if we change the old connector version

File: test/lib/commonkafka.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def delete_kafka_connector(setup, connector):
     response = requests.delete(url=setup["kafka_connect_url"] + "/connectors/" + connector,
                                headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
     if response.status_code == 204:
-        logger.info("Deleted connector successfully - " + connector)
+        # logger.info("Deleted connector successfully - " + connector)
```
Comment (chaitanyaphalak) [inline test/lib/commonkafka.py:77]: can be changed to debug

**Review by chaitanyaphalak — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_build_test.yaml [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    strategy:
+      matrix:
+        include:
+          - kafka_version: "1.1.1"
+            kafka_package: "kafka_2.11-1.1.1.tgz"
+          - kafka_version: "2.0.0"
+            kafka_package: "kafka_2.11-2.0.0.tgz"
+          - kafka_version: "2.1.0"
+            kafka_package: "kafka_2.12-2.1.0.tgz"
+          - kafka_version: "2.6.0"
```
[the remark this answers is not in the document]
Comment (mwang2016) [reply .github/workflows/ci_build_test.yaml:54]: sure, i will add them

File: .github/workflows/ci_build_test.yaml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            kafka_package: "kafka_2.13-2.6.0.tgz"
+    env:
+      CI_SPLUNK_VERSION: "8.2.0"
```
[the remark this answers is not in the document]
Comment (mwang2016) [reply .github/workflows/ci_build_test.yaml:57]: sure

File: .github/workflows/ci_build_test.yaml [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+      CI_SPLUNK_PORT: 8089
+      CI_SPLUNK_USERNAME: admin
+      CI_SPLUNK_HEC_TOKEN: a6b5e77f-d5f6-415a-bd43-930cecb12959
+      CI_SPLUNK_HEC_TOKEN_ACK: a6b5e77f-d5f6-415a-bd43-930cecb12950
+      CI_SPLUNK_PASSWORD: helloworld
+      CI_INDEX_EVENTS: main
+      CI_KAFKA_HEADER_INDEX: kafka
+      CI_DATAGEN_IMAGE: rock1017/log-generator:latest
+      CI_OLD_CONNECTOR_VERSION: v1.2.0
+      KUBERNETES_VERSION: v1.15.2
```
[the remark this answers is not in the document]
Comment (mwang2016) [reply .github/workflows/ci_build_test.yaml:69]: Yes, it was a mistake, I thought i would install splunk on kubernetes, forgot to remove it after.

File: .github/workflows/ci_build_test.yaml [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+      CI_SPLUNK_HOST: 127.0.0.1
+      CI_SPLUNK_PORT: 8089
+      CI_SPLUNK_USERNAME: admin
+      CI_SPLUNK_HEC_TOKEN: a6b5e77f-d5f6-415a-bd43-930cecb12959
+      CI_SPLUNK_HEC_TOKEN_ACK: a6b5e77f-d5f6-415a-bd43-930cecb12950
+      CI_SPLUNK_PASSWORD: helloworld
+      CI_INDEX_EVENTS: main
+      CI_KAFKA_HEADER_INDEX: kafka
+      CI_DATAGEN_IMAGE: rock1017/log-generator:latest
+      CI_OLD_CONNECTOR_VERSION: v1.2.0
```
[the remark this answers is not in the document]
Comment (mwang2016) [reply .github/workflows/ci_build_test.yaml:68]: It is the old version of connector that I need to use for testing upgrade. The idea is to install v1.2.0 first, start sending data and then upgrade to v2.0.2

File: test/lib/commonkafka.py (hunk 2 above)
Comment (mwang2016) [reply test/lib/commonkafka.py:77]: yes

**Review by chaitanyaphalak — DISMISSED** [review]

**Review by chaitanyaphalak — APPROVED** [review]

**Pull request merged**